src/core/Table/filters/DateRangeFilter/DateRangeFilter.tsx

```diff
@@ -0,0 +1,110 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useTheme } from '../../../utils/hooks/useTheme';
+import {
+  FilterButtonBar,
+  FilterButtonBarTranslation,
+} from '../FilterButtonBar';
+import { BaseFilter } from '../BaseFilter';
+import { TableFilterProps } from '../types';
+import '@itwin/itwinui-css/css/table.css';
+import DatePickerInput from './DatePickerInput';
+
+export type DateRangeTranslation = {
+  from: string;
+  to: string;
+};
+
+const defaultStrings: DateRangeTranslation = {
+  from: 'From',
+  to: 'To',
+};
+
+const defaultFormatDate = (date: Date) => {
+  const formatter = new Intl.DateTimeFormat('en-us', {
+    month: 'short',
+    day: 'numeric',
+    year: 'numeric',
+  });
+  return formatter.format(date);
+};
+
+const defaultParseInput = (text: string) => {
+  // Checks whether date is in correct format e.g. Feb 11, 2021
+  if (/^[a-z]{3}\s\d{1,2},\s\d{4}$/gi.test(text)) {
+    return new Date(text);
```

Comment:
I was trying to research how best to parse dates and I came across this (notice emphasis on "strongly" 😄)

![image](https://user-images.githubusercontent.com/9084735/118821007-d0034080-b884-11eb-895e-fea8fc99ca15.png)

I guess it's fine here but we should make it extra clear that we expect users to provide their own parser

Reply:
Other option would be to use some date parsing lib like `date-fns` but native parse with our date format worked perfectly to my surprise.

Reply:
Yeah no, I think it's fine how we are doing it but we should also explicitly state that we expect users to handle this if they want reliable results.

Reply:
Added this comment to `tableFilters`:
![image](https://user-images.githubusercontent.com/36186912/118988337-c6afcc00-b989-11eb-9fb4-20e51041fb38.png)